### ISSUE TITLE
Unable to detect a lost connection

### ISSUE THREAD

**Issue by moafred** [Evaneos/Burrow#22]
Environnement:
Message Broker: RabbitMQ
Driver: php-amqplib
Hosting: GKE
Orchestration: Kubernetes
Containerization: Docker

We have 2 workers, a publisher and a consumer.

After about 8 hours of inactivity by the publisher and when it start again to push messages into RabbitMQT, the container got an error: `syslog: write tcp x.x.x.x:41132->x.x.x.x:5000: write: broken pipe`.
No problem, the app crash and the container is restarted by Kubernetes.

But for 1 day by week the consumer did not have this error (that the publisher always have).
In this case, the consumer stay alive without consuming the new messages because the connection is lost but he is not aware of it.

We have updated our container by using the amqp c library instead and let know here if we still have this issue or not.

**Comment by remi-san** [comment]
I'm wondering if this broken pipe is not a docker issue... but I don't know why there's a different behavior between those two containers.

**Comment by moafred** [comment]
```
I'm wondering if this broken pipe is not a docker issue... 
```
I think it's specific of the Docker+GKE environment.

```
but I don't know why there's a different behavior between those two containers.
```
Exactly, don't know too.

**Comment by moafred** [comment]
After updating our app in order to use the C lib, we still have the issue.

It turns that the `broken pipe` errors are from a deprecated use of logspout.

Here the logs:
<img width="1055" alt="screen shot 2017-12-04 at 09 55 59" src="https://user-images.githubusercontent.com/1343723/33548869-f5d6f79a-d8e8-11e7-9c61-06cb2dfe20e5.png">

There is 3 RabbitMq errors `client unexpectedly closed TCP connection`, which are associated with our 3 Consumer workers.

We continue to investigate.

**Comment by moafred** [comment]
Implementing Heartbeats seems to be a good option: https://www.rabbitmq.com/heartbeats.html

**Comment by remi-san** [comment]
Yup, I'll try to give you access to it through Burrow then... via an isAlive() on the driver, or something like this...
